enrichments/trace/internal/elastic/span_test.go

```diff
@@ -375,6 +375,60 @@ func TestElasticSpanEnrich(t *testing.T) {
 				AttributeServiceTargetName: "testsvc",
 			},
 		},
+		{
+			name: "db_over_http",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutStr(
+					semconv.AttributeURLFull,
+					"https://localhost:9200/index/_search?q=user.id:kimchy",
+				)
+				span.Attributes().PutStr(
+					semconv.AttributeDBSystem,
+					semconv.AttributeDBSystemElasticsearch,
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "elasticsearch",
+				AttributeServiceTargetName: "testsvc",
+			},
+		},
+		{
+			name: "db_over_rpc",
+			input: func() ptrace.Span {
+				span := getElasticSpan()
+				// peer-service should be ignored if more specific deductions
+				// can be made about the service target.
+				span.Attributes().PutStr(semconv.AttributePeerService, "testsvc")
+				span.Attributes().PutStr(
+					semconv.AttributeRPCSystem,
+					semconv.AttributeRPCSystemGRPC,
+				)
+				span.Attributes().PutStr(semconv.AttributeRPCService, "cassandra.API")
+				span.Attributes().PutStr(
+					semconv.AttributeRPCGRPCStatusCode,
+					semconv.AttributeRPCGRPCStatusCodeOk,
+				)
+				span.Attributes().PutStr(
+					semconv.AttributeDBSystem,
+					semconv.AttributeDBSystemCassandra,
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeServiceTargetType: "cassandra",
+				// TODO (lahsivjar): Current apm-data would not fallback to using
+				// RPC/HTTP when a database span is detected but all fields are not
+				// populated. Should it?
+				AttributeServiceTargetName: "testsvc",
```

Comment:
@gregkalapos @axw What are your thoughts on this? As in this test case, we won't fallback to setting the svc target attributes in cases where span represents a db type over HTTP BUT the db attributes are not enough to infer all required information.

Reply:
I would be inclined to go with bug-for-bug compatibility here for now, and create an issue and follow up. I think it probably makes sense to have an order of precedence like DB > RPC > HTTP > general, both for setting the target type and target name. OTOH it's kinda awkward that one attribute would be set from DB, and another from RPC/HTTP, so it's not 100% obvious to me.

Reply:
➕ on the suggestion. I think the core issue is that some of the fields we use here are not required (e.g. `db.name`), so this is not only a theoretical problem - this can happen while the data is 100% compatible with SemConv.

If there is an issue, I'm happy to discuss how we could improve it.